pkg/ingester/ingester.go

```diff
@@ -495,6 +536,58 @@ func (i *Ingester) append(ctx context.Context, userID string, labels labelPairs,
 	return err
 }
 
+func (i *Ingester) appendMetadata(ctx context.Context, userID string, metadata *client.MetricMetadata) error {
+	i.userStatesMtx.RLock()
+	if i.stopped {
+		i.userStatesMtx.RUnlock()
+		return fmt.Errorf("ingester stopping")
+	}
+	i.userStatesMtx.RUnlock()
+
+	i.usersMetadataMtx.Lock()
+	defer i.usersMetadataMtx.Unlock()
+
+	// As we get the set, we also validate two things:
+	// 1. The user is allowed to create new metrics to add metadata to.
+	// 2. If the metadata set is already present, it hasn't reached the limit of metadata we can append.
+	userMetadata := i.getOrCreateUserMetadata(userID)
+	set, ok := userMetadata.get(metadata.GetMetricName())
+	if !ok {
+		// Verify that the user can create more metric metadata given we don't have a set for that metric name.
+		if err := i.limiter.AssertMaxMetricsWithMetadataPerUser(userID, userMetadata.length()); err != nil {
+			validation.DiscardedMetadata.MustCurryWith(prometheus.Labels{"user": userID}).WithLabelValues(perUserMetadataLimit).Inc()
```

Comment:
Why not simply `validation.DiscardedMetadata.WithLabelValues(perUserMetadataLimit, userID).Inc()` ?